include/swift/Basic/ParseableOutput.h

```diff
@@ -0,0 +1,76 @@
+//===----- ParseableOutput.h - Helpers for parseable output ------- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// Helpers for emitting the compiler's parseable output.
+///
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_PARSEABLEOUTPUT_H
+#define SWIFT_PARSEABLEOUTPUT_H
+
+#include "swift/Basic/LLVM.h"
+#include "swift/Basic/TaskQueue.h"
+#include "swift/Basic/FileTypes.h"
+
+namespace swift {
+
+namespace parseable_output {
+
+/// Quasi-PIDs are _negative_ PID-like unique keys used to
+/// masquerade batch job constituents as (quasi)processes, when writing
+/// parseable output to consumers that don't understand the idea of a batch
+/// job. They are negative in order to avoid possibly colliding with real
+/// PIDs (which are always positive). We start at -1000 here as a crude but
+/// harmless hedge against colliding with an errno value that might slip
+/// into the stream of real PIDs (say, due to a TaskQueue bug).
+const int QUASI_PID_START = -1000;
+
+struct CommandInput {
+  std::string Path;
+  CommandInput() {}
+  CommandInput(StringRef Path) : Path(Path) {}
+};
+
+using OutputPair = std::pair<file_types::ID, std::string>;
+
+struct DetailedMessagePayload {
+  std::string Executable;
+  SmallVector<std::string, 16> Arguments;
+  std::string CommandLine;
+  SmallVector<CommandInput, 4> Inputs;
+  SmallVector<OutputPair, 8> Outputs;
+};
```

Comment:
This type ought to have a doc comment.

I'd also like you to think carefully about its name. Yes, from the perspective of the `emit*Message()` functions, this is the payload of the `DetailedMessage` you're emitting. But that's an implementation detail of this subsystem.

A better type name would come from asking, "what is this type to the *clients* of this header"? I might call it something like a "job description" or "process info" or maybe a "quasi-invocation" or something. I'll leave the exact decision to you.

Reply:
Thank you for the insight, I fully agree. 
I have renamed the type and it now has a doc to describe its purpose.